### PR TITLE
Added the ability to specify a 'team' in the Slack backend and add in the ID field if it's not present. Fixes #33 #35

### LINKED ISSUE #33: Add the ability to enforce 'team' restriction on Slack auth
Simple addition of a param to pass to the Slack auth URL.

### PR DESCRIPTION
**Pull request by Schwanksta**

### CODE REVIEW

**Comment by Schwanksta** [comment]
I updated this pull request to reflect a bug I came across trying to implement Slack auth. I am not sure if Slack sends different JSON formats for different API calls, but the default call didn't return in the UID in a way we could pull it back out.

File: social_core/backends/slack.py [hunk 1]
```diff
@@ -21,6 +21,12 @@ class SlackOAuth2(BaseOAuth2):
         ('real_name', 'real_name')
     ]
 
+    def auth_extra_arguments(self):
+        params = super(SlackOAuth2, self).auth_extra_arguments() or {}
+        if self.setting('team', None):
```
Comment (omab) [inline social_core/backends/slack.py]: Setting names should be all uppercase.

Comment (omab) [inline social_core/backends/slack.py]: I've fixed and applied PEP8.

**Comment by omab** [comment]
Thanks!

**Pull request merged**